src/SALib/analyze/delta.py

```diff
@@ -15,12 +15,15 @@ def analyze(
     conf_level: float = 0.95,
     print_to_console: bool = False,
     seed: int = None,
+    Nr: int = None,
+    analyze_sobol: bool = True,

```

Comment:
I would call it `method` which could get `"all", "sobol", "delta"`. That would be more flexible for future improvements I think.

Reply:
I propose to completely remove sobol indices from this method and move it to a different file. It is simply not related to DMIM. Although a different implementation of sobol indices already exists in SALib.

Reply:
I replied above. But in the literature you do see expressions of Sobol' indices with moment independent. So this is just a different metric. I do agree that the name of the function is not general enough then to convey that. But we cannot change like this the parameters and features.

What we can do though is (in some follow up) to think about whether we would want or not to have a new API that would be more general. Doing so, we would be able to rework the API as we want and then ultimately deprecate this function.